.github/workflows/upload.yml

```diff
@@ -0,0 +1,163 @@
+name: Upload
+on:
+  release:
+    types: [published]
+
+env:
+  TF_VERSION: 2.6
+  TORCH_VERSION: 1.10.0+cpu
+  COVERAGE_FLAGS: "--cov=pennylane --cov-report=term-missing --cov-report=xml --no-flaky-report -p no:warnings --tb=native"
+
+
+jobs:
+
+  core-tests:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        config:
+          - { python-version: 3.7, interfaces: [ 'torch' ] }
+          - { python-version: 3.7, interfaces: [ 'tf' ] }
+          - { python-version: 3.8, interfaces: [ 'tf', 'torch', 'jax' ] }
+          - { python-version: 3.9, interfaces: [ ] }
+          - { python-version: '3.10', interfaces: [ ] }
+
+
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.4.1
+        with:
+          access_token: ${{ github.token }}
+
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
```

Comment:
Why this depth?

Reply:
Copy of original test suite. Added in https://github.com/PennyLaneAI/pennylane/commit/85dd19333a71907f2b5bfd9e1e928ecefd9ab7d4.

Reply:
I can remove it then !